parcels/field.py

```diff
@@ -994,33 +988,34 @@ def subset(self, dim, dimname, indices):
             elif len(dim.shape) == 4:  # SGrid
                 inds_depth = indices['depth'] if 'depth' in indices else range(dim.shape[0])
                 dim_inds = dim[:, inds_depth, inds_lat, inds_lon]
-        return dim_inds
+        return np.array(dim_inds)
 
     def read_lonlat(self, indices):
-        lon = self.dataset[self.dimensions['lon']]
-        lat = self.dataset[self.dimensions['lat']]
-        if len(lon.shape) > 1:
-            londim = lon.shape[0]
-            latdim = lat.shape[1]
-            if np.allclose(lon[0, :], lon[int(londim/2), :]) and np.allclose(lat[:, 0], lat[:, int(latdim/2)]):
-                lon = lon[0, :]
-                lat = lat[:, 0]
+        lon = getattr(self.dataset, self.dimensions['lon'])
+        lat = getattr(self.dataset, self.dimensions['lat'])
         lon_subset = self.subset(lon, 'lon', indices)
         lat_subset = self.subset(lat, 'lat', indices)
+        if len(lon_subset.shape) > 1:
+            londim = lon_subset.shape[0]
+            latdim = lat_subset.shape[1]
+            if np.allclose(lon_subset[0, :], lon_subset[int(londim/2), :]) and np.allclose(lat_subset[:, 0], lat_subset[:, int(latdim/2)]):
+                lon_subset = lon_subset[0, :]
+                lat_subset = lat_subset[:, 0]
         return lon_subset, lat_subset
 
     def read_depth(self, indices):
         if 'depth' in self.dimensions:
-            depth = self.dataset[self.dimensions['depth']]
+            depth = getattr(self.dataset, self.dimensions['depth'])
             return self.subset(depth, 'depth', indices)
         else:
             return np.zeros(1)
 
     @property
     def data(self):
-        if len(self.dataset[self.name].shape) == 2:
-            data = self.dataset[self.name][self.indslat, self.indslon]
-        elif len(self.dataset[self.name].shape) == 3:
+        data = getattr(self.dataset, self.name)
+        if len(data.shape) == 2:
+            data = data[self.indslat, self.indslon]
+        elif len(data.shape) == 3:
             data = self.dataset[self.name][:, self.indslat, self.indslon]
```

Comment:
Also, what happens if field is in `(depth, lat, lon)`, instead of `(time, lat, lon)`? Shouldn't we be using `self.indsdepth` too, then? Note that in the previous version this wasn't possible either, and we might get rid of explicit indexing in the near future, so if there's no easy fix then probably better to leave as is for now